src/common/payment.ts

```diff
@@ -207,4 +269,110 @@ export class TurboAuthenticatedPaymentService
       await this.signer.generateSignedRequestHeaders(),
     );
   }
+
+  private async getTargetWalletForFund(): Promise<string> {
+    const { addresses } = await this.httpService.get<TurboInfoResponse>({
+      endpoint: '/info',
+    });
+
+    return addresses[this.token];
+  }
+
+  // TODO: token based backoff
+  private async pollForTxBeingAvailableFromGateway(txId: string) {
+    const maxAttempts = 30;
+    const pollingIntervalMs = 1_000;
+    const initialBackoffMs = 5_000;
+
+    await new Promise((resolve) => setTimeout(resolve, initialBackoffMs));
+
+    let attempts = 0;
+    while (attempts < maxAttempts) {
+      const response = await this.arweave.api.post('/graphql', {
+        query: `
+          query {
+            transaction(id: "${txId}") {
+              recipient
+              owner {
+                address
+              }
+              quantity {
+                winston
+              }
+            }
+          }
+        `,
+      });
```

Comment:
should we `catch` errors here and increment the attempt count? the calling function is not wrapped in the try/catch

Reply:
a324ca1